### PR TITLE
RTD bugfix

### PR DESCRIPTION
**Pull request by rstoneback**
# Description

Addresses # (issue)

Read the Docs documentation is not building due to a lack of h5py. Fortran packages are currently excluded from setup.py requirements as RTD (at least historically) does not support building Fortran packages.

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Verified the documentation builds on RTD

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``master``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Comment by rstoneback** [comment]
Making a pull for visibility as I'll be in and out today. Starting with the simplest fix so I can get the branch into RTD and see how it handles over there.

I've seen notes that there is a bug for mocked packages in RTD v3.0. That likely drives the recommendation for the .yml file that I've added. If RTD can't build Fortran though then adding the package to requirements won't help. I've started out down the recommended path.

**Comment by rstoneback** [comment]
I checked, including the fortran packages did not solve the issue.

File: .readthedocs.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+formats:
+  - pdf
+  - epub
+
+# Optionally set the version of Python and requirements
+# required to build your docs
+python:
+  version: 3.7
+  install:
+    - requirements: docs/requirements.txt
```
Comment (aburrell) [inline .readthedocs.yml:22]: Looks like yml wants a newline at the end of the file

**Comment by rstoneback** [comment]
Yes, I’ll include that. Thanks.

**Comment by rstoneback** [comment]
I accounted for @aburrell comments. I found a few other areas not included in the API, then I noticed an issue with organization. I tried to sort it out. I think all the changes within are good, except for 1. The templates part of the API fails with a can't import pysat.instruments.templates. I'm not sure why it used to work, even in this branch, but it doesn't now. 

```
WARNING: autodoc: failed to import module u'templates.template_instrument' from module u'pysat.instruments'; the following exception was raised:
No module named templates

```

I'm getting ready for the dentist otherwise I'd continue immediately on the last little bit. I'll be back later.

**Comment by rstoneback** [comment]
> I accounted for @aburrell comments. I found a few other areas not included in the API, then I noticed an issue with organization. I tried to sort it out. I think all the changes within are good, except for 1. The templates part of the API fails with a can't import pysat.instruments.templates. I'm not sure why it used to work, even in this branch, but it doesn't now.
> 
> ```
> WARNING: autodoc: failed to import module u'templates.template_instrument' from module u'pysat.instruments'; the following exception was raised:
> No module named templates
> ```
> 
> I'm getting ready for the dentist otherwise I'd continue immediately on the last little bit. I'll be back later.

It works as well as it ever did on RTD. Methods for most things are showing up under templates.

**Pull request merged**